tests/parallel_test_core.py

```diff
@@ -27,10 +27,21 @@ def run(self):
 # run tests for one mode
 def run_mode(args):
   mode = args[0]
-  print '<< running %s >>' % mode
+  if len(args) > 1:
+    # If args has multiple elements, then only the selected tests from that suite are run e.g. args=['asm1', 'test_hello_world', 'test_i64'] runs only tests asm1.test_hello_world and asm1.test_i64.
+    for i in range(1, len(args)):
+      if args[i].startswith('test_'):
+        args[i] = mode + '.' + args[i]
+    args = args[1:]
```

Comment:
this pull lgtm. @aidanhs's note here is a possible way to make the code prettier, but up to you @juj